fapolicy_analyzer/ui/faprofiler.py

```diff
@@ -226,6 +242,112 @@ def clean_all(self):
+        """
+        Validates the Profiler Session object's user, target, pwd parameters.
+        Returns a dictionary mapping enums to error msgs.
+        """
+        dictReturn = {}
+        logging.debug(f"validateProfArgs({dictProfTgt}")
+        dictReturn[ProfSessionArgsStatus.OK] = s.PROF_ARG_OK
+        error_msg = s.PROF_ARG_OK
+
+        exec_path = dictProfTgt["executeText"]
+        exec_user = dictProfTgt["userText"]
+        exec_pwd = dictProfTgt["dirText"]
+
+        # exec empty?
+        if not exec_path:
+            error_msg = s.PROF_ARG_EXEC_EMPTY
+            if ProfSessionArgsStatus.OK in dictReturn:
+                dictReturn.pop(ProfSessionArgsStatus.OK)
+            dictReturn[ProfSessionArgsStatus.EXEC_EMPTY] = error_msg
+
+        # exist?
+        elif not os.path.exists(exec_path):
+            if ProfSessionArgsStatus.OK in dictReturn:
+                dictReturn.pop(ProfSessionArgsStatus.OK)
+            error_msg = exec_path + s.PROF_ARG_EXEC_DOESNT_EXIST
+            dictReturn[ProfSessionArgsStatus.EXEC_DOESNT_EXIST] = error_msg
+
+        # executable?
+        elif not os.access(exec_path, os.X_OK):
+            if ProfSessionArgsStatus.OK in dictReturn:
+                dictReturn.pop(ProfSessionArgsStatus.OK)
+            error_msg = exec_path + s.PROF_ARG_EXEC_NOT_EXECUTABLE
+            dictReturn[ProfSessionArgsStatus.EXEC_NOT_EXECUTABLE] = error_msg
+
+        if ProfSessionArgsStatus.OK in dictReturn:
+            logging.debug("FaProfSession::validateArgs() --> exec verified")
+
+        # user?
+        try:
+            if exec_user:
+                pwd.getpwnam(exec_user)
+        except KeyError as e:
+            logging.debug(f"User {exec_user} does not exist: {e}")
+            error_msg = exec_user + s.PROF_ARG_USER_DOESNT_EXIST
+            if ProfSessionArgsStatus.OK in dictReturn:
+                dictReturn.pop(ProfSessionArgsStatus.OK)
+            dictReturn[ProfSessionArgsStatus.USER_DOESNT_EXIST] = error_msg
+
+        if ProfSessionArgsStatus.OK in dictReturn:
+            logging.debug("FaProfSession::validateArgs() --> user verified")
+
+        # working dir?
+        # pwd empty?
+        if exec_pwd:
+            if not os.path.exists(exec_pwd):
+                error_msg = exec_pwd + s.PROF_ARG_PWD_DOESNT_EXIST
+                if ProfSessionArgsStatus.OK in dictReturn:
+                    dictReturn.pop(ProfSessionArgsStatus.OK)
+                dictReturn[ProfSessionArgsStatus.PWD_DOESNT_EXIST] = error_msg
+
+            elif not os.path.isdir(exec_pwd):
+                error_msg = exec_pwd + s.PROF_ARG_PWD_ISNT_DIR
+                if ProfSessionArgsStatus.OK in dictReturn:
+                    dictReturn.pop(ProfSessionArgsStatus.OK)
+                dictReturn[ProfSessionArgsStatus.PWD_ISNT_DIR] = error_msg
+
+        if ProfSessionArgsStatus.OK in dictReturn:
+            logging.debug("FaProfSession::validateArgs() --> pwd verified")
+
+        return dictReturn
```

Comment:
```suggestion
        dictReturn = {}
        logging.debug(f"validateProfArgs({dictProfTgt}")

        exec_path = dictProfTgt["executeText"]
        exec_user = dictProfTgt["userText"]
        exec_pwd = dictProfTgt["dirText"]

        # exec empty?
        if not exec_path:
            dictReturn[ProfSessionArgsStatus.EXEC_EMPTY] = s.PROF_ARG_EXEC_EMPTY

        # exist?
        elif not os.path.exists(exec_path):
            dictReturn[ProfSessionArgsStatus.EXEC_DOESNT_EXIST] = (
                exec_path + s.PROF_ARG_EXEC_DOESNT_EXIST
            )

        # executable?
        elif not os.access(exec_path, os.X_OK):
            dictReturn[ProfSessionArgsStatus.EXEC_NOT_EXECUTABLE] = (
                exec_path + s.PROF_ARG_EXEC_NOT_EXECUTABLE
            )

        if not dictReturn:
            logging.debug("FaProfSession::validateArgs() --> exec verified")

        # user?
        try:
            if exec_user:
                pwd.getpwnam(exec_user)
        except KeyError as e:
            logging.debug(f"User {exec_user} does not exist: {e}")
            dictReturn[ProfSessionArgsStatus.USER_DOESNT_EXIST] = (
                exec_user + s.PROF_ARG_USER_DOESNT_EXIST
            )

        if not dictReturn:
            logging.debug("FaProfSession::validateArgs() --> user verified")

        # working dir?
        # pwd empty?
        if exec_pwd:
            if not os.path.exists(exec_pwd):
                dictReturn[ProfSessionArgsStatus.PWD_DOESNT_EXIST] = (
                    exec_pwd + s.PROF_ARG_PWD_DOESNT_EXIST
                )

            elif not os.path.isdir(exec_pwd):
                dictReturn[ProfSessionArgsStatus.PWD_ISNT_DIR] = (
                    exec_pwd + s.PROF_ARG_PWD_ISNT_DIR
                )

        if not dictReturn:
            logging.debug("FaProfSession::validateArgs() --> pwd verified")

        return dictReturn or {ProfSessionArgsStatus.OK: s.PROF_ARG_OK}
```

Reply:
Beyond my suggestion above for this method, I'm not sure I'm seeing the point of returning a dict with the Enum as keys. I can't see where you ever check the key value other than to see if its `ProfSessionArgsStatus.OK`. Wouldn't it be easier to just return a list of error messages here and if the list is empty then validation is `OK`?

Reply:
The idea is to provide both the offending field and error text to the UI layer so that if the UI layer wants to also highlight the text field (background color, red border, whatevver) it can. The text provides a file specific msg suitable for the pop-up.

I will review your suggestions and update you hopefully later today.

Thx @dorschs57 !

Reply:
Used your suggested change above.